### PR TITLE
:sparkles: Add Method to Request a Video's Transcript

### PR DESCRIPTION
**Pull request by SulimanLab**
this PR add an API to get the video's transcript with examples

### CODE REVIEW

**Comment by tombulled** [comment]
Hi @SulimanLab, thanks for opening this PR these changes look really good!

File: innertube/enums.py [hunk 1]
```diff
@@ -14,6 +14,7 @@ class Endpoint(StrEnum):
     NEXT: str = "next"
     MUSIC_GET_SEARCH_SUGGESTIONS: str = "music/get_search_suggestions"
     MUSIC_GET_QUEUE: str = "music/get_queue"
+    CAPTION: str = "get_transcript"
```
Comment (tombulled) [inline innertube/enums.py:17]: Although naming this `CAPTION` makes sense, for consistency I'd prefer this to be named after the endpoint itself. Therefore `GET_TRANSCRIPT` would be the preferred name for this enum member

File: innertube/clients.py [hunk 2]
```diff
@@ -97,12 +97,35 @@ def player(self, video_id: str) -> dict:
             ),
         )
 
+    def caption(
```
Comment (tombulled) [inline innertube/clients.py:100]: Although `caption` makes sense here too, for consistency it'd be preferable for this method to be renamed to `get_transcript`

File: innertube/clients.py [hunk 3]
```diff
@@ -97,12 +97,35 @@ def player(self, video_id: str) -> dict:
...
+            self,
+            browse_id: Optional[str] = None,
```
Comment (tombulled) [inline innertube/clients.py:102]: From some initial research, I am only seeing the `params` field being used for the `get_transcript` API call.
For example, here's a minimal POC:
```python
from innertube import InnerTube

client = InnerTube("WEB")

transcript = client(
    "get_transcript",
    body={
        "params": "CgtkUXc0dzlXZ1hjURISQ2dOaGMzSVNBbVZ1R2dBJTNEGAEqM2VuZ2FnZW1lbnQtcGFuZWwtc2VhcmNoYWJsZS10cmFuc2NyaXB0LXNlYXJjaC1wYW5lbDABOAE%3D",
    },
)
```

Your changes appear to add `browse_id` and `continuation` parameters. Have you observed these being used in the wild for this API endpoint?

Comment (tombulled) [reply innertube/clients.py:102]: Even for incredibly long videos (e.g. >2hrs) YouTube does not appear to make use of continuations for the transcript from what I can see

**Comment by tombulled** [comment]
Interestingly (from some very basic testing), the `get_transcript` endpoint only appears to work for YouTube clients.

These are the clients I tested:
| Service | Client | Success | Description |
| ------- | ------------ | ----------- | -------------- |
| YouTube | WEB | Yes | JSON Response |
| YouTube Music | WEB_REMIX | No | 400 Bad Request: Precondition check failed. |
| YouTube Kids | WEB_KIDS | No | 400 Bad Request: Precondition check failed. |
| YouTube Studio | WEB_CREATOR | No | 400 Bad Request: Precondition check failed. |

File: innertube/clients.py (hunk 3 above)
Comment (tombulled) [reply innertube/clients.py:102]: Under the assumption that `params` is the only parameter needed, the method should be able to become as simple as:
```python
def get_transcript(self, params: str) -> dict:
    return self(
        Endpoint.GET_TRANSCRIPT,
        body=dict(
            params=params,
        ),
    )
```

**Comment by BRUHItsABunny** [comment]
Does the output from this API call differ from YT's timedtext API?
image

**Comment by tombulled** [comment]
@BRUHItsABunny they look very similar from what I can tell, very interesting find!

**Comment by tombulled** [comment]
@SulimanLab Once again, thanks for opening this PR. Unfortunately as the comments have not been addressed, I'm going to go ahead and close this PR. I'll open a separate one with the necessary changes made :slightly_smiling_face:

**Comment by BRUHItsABunny** [comment]
Yea I was just wondering since I currently rely on the timed text api to download video subs/captions

**Comment by tombulled** [comment]
I'll get a new version of `innertube` released today with the `get_transcript` method implemented, so hopefully might come in handy

**Comment by tombulled** [comment]
`innertube` v2.1.3 released as of #43

**Pull request closed without merging**